src/application.coffee

```diff
@@ -136,6 +149,21 @@ application.kill = kill = (app, updateDB = true, removeContainer = true) ->
 		logSystemEvent(logTypes.stopAppError, app, err)
 		throw err
 
+application.deleteImage = deleteImage = (app) ->
+	logSystemEvent(logTypes.deleteImageForApp)
+	docker.getImage(app.imageId).removeAsync(force: true)
+	.then ->
+		logSystemEvent(logTypes.deleteImageForAppSuccess, app)
+	.catch Promise.OperationalError, (err) ->
```

Comment:
You should be able to do
```coffee
.catch statusCode: 404, (err) ->
	logSystemEvent(logTypes.imageAlreadyDeleted, app)
.catch (err) ->
	logSystemEvent(logTypes.deleteImageForAppError, app, err)
	throw err
```
to make this a lot nicer, alternatively if the string forcing is needed:
```coffee
ImageNotFoundError = (err) ->
	return "#{err.statusCode}" is '404'
...
.catch ImageNotFoundError, (err) ->
	logSystemEvent(logTypes.imageAlreadyDeleted, app)
.catch (err) ->
	logSystemEvent(logTypes.deleteImageForAppError, app, err)
	throw err
```